server/store/memory/peer_store.go

```diff
@@ -88,16 +90,19 @@ func leechersKey(infoHash chihaya.InfoHash) string {
 
 func (s *peerStore) PutSeeder(infoHash chihaya.InfoHash, p chihaya.Peer) error {
 	key := seedersKey(infoHash)
+	i := s.shardIndex(infoHash)
+	if s.shards[i] == nil {
```

Comment:
Better initialize all shards in `New` already, easier to read and saves the `nil` checks on every call